src/collections/config/types/generative.ts

```diff
@@ -75,8 +75,9 @@ export type GenerativeOpenAIConfig = GenerativeOpenAIConfigBase & {
 };
 
 export type GenerativeAzureOpenAIConfig = GenerativeOpenAIConfigBase & {
-  resourceName: string;
-  deploymentId: string;
+  resourceName?: string;
+  deploymentId?: string;
+  isAzure?: true;
```

Comment:
If this is always true, does it still need the ? operator? If not, can we remove it?

Reply:
same as https://github.com/weaviate/typescript-client/pull/196#discussion_r1771153555 🤔

Reply:
Replied below 😁